### PR TITLE
Support new scene bundle and refactor material

### PR DESCRIPTION
**Pull request by zhuxudong**
### Please check if the PR fulfills these requirements

- [x] The commit message follows our [guidelines](https://github.com/galacean/engine/blob/main/.github/COMMIT_MESSAGE_CONVENTION.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)
image
image

### Other information:

### CODE REVIEW

**Comment by zhuxudong** [comment]
#1351

File: packages/loader/src/MaterialLoader.ts [hunk 1]
```diff
@@ -1,52 +1,39 @@
...
+  if (typeof value === "object") {
+    for (let subKey in value) {
+      setProperty(object[key], subKey, value[subKey]);
+    }
+  } else {
+    object[key] = value;
+  }
+}
 
 @resourceLoader(AssetType.Material, ["json"])
```
Comment (GuoLei1990) [inline packages/loader/src/MaterialLoader.ts:25]: The extension use  `json` is not good, maybe should use `material`

File: packages/loader/src/resource-deserialize/resources/prefab/PrefabDesign.ts [hunk 2]
```diff
@@ -63,3 +86,16 @@ export type IClassObject = {
...
+  name: string;
+  shader: string;
+  shaderData: {
+    [key: string]: {
+      type: "Vector2" | "Vector3" | "Vector4" | "Color" | "Float" | "Texture";
+      value: any;
+    };
+  };
+  macros: Array<{ name: string; value?: string }>;
+  renderState: RenderState;
```
Comment (GuoLei1990) [inline packages/loader/src/resource-deserialize/resources/prefab/PrefabDesign.ts:100]: Sub-attributes need to be completed

File: packages/loader/src/MaterialLoader.ts [hunk 3]
```diff
@@ -1,52 +1,39 @@
...
-  load(item: LoadItem, resourceManager: ResourceManager): AssetPromise<string> {
+class MaterialLoader extends Loader<Material> {
+  load(item: LoadItem, resourceManager: ResourceManager): AssetPromise<Material> {
     return new AssetPromise((resolve, reject) => {
       this.request(item.url, {
         ...item,
         type: "json"
       })
-        .then((json: { [key: string]: any }) => {
+        .then((json: IPrefabMaterial) => {
```
Comment (GuoLei1990) [inline packages/loader/src/MaterialLoader.ts:33]: `json`->`materialSchema`

File: packages/loader/src/MaterialLoader.ts [hunk 4]
```diff
@@ -1,52 +1,39 @@
...
-            case "bake-pbr":
-              // @todo refactor custom shader later
-              // @ts-ignore
-              material = new PBRBaseMaterial(engine, Shader.find("bake-pbr"));
-              break;
-          }
+          const material = new Material(engine, Shader.find(shader));
           material.name = name;
 
           const texturePromises = new Array<Promise<Texture2D | void>>();
```
Comment (GuoLei1990) [inline packages/loader/src/MaterialLoader.ts:39]: `new Array<Promise<Texture2D | void>>()` or `new Array<Promise<Texture2D>>()`?

File: packages/loader/src/resource-deserialize/resources/prefab/PrefabDesign.ts [hunk 5]
```diff
@@ -63,3 +86,16 @@ export type IClassObject = {
 export type IBasicType = string | number | boolean | null | undefined | IAssetRef | IClassObject | IMethodParams;
 
 export type IAssetRef = { key?: string; refId: string };
+
+export interface IPrefabMaterial {
```
Comment (GuoLei1990) [inline packages/loader/src/resource-deserialize/resources/prefab/PrefabDesign.ts:90]: This should named `IMaterialSchema` or `IMaterialDefine`

File: packages/loader/src/resource-deserialize/resources/prefab/PrefabDesign.ts [hunk 6]
```diff
@@ -63,3 +86,16 @@ export type IClassObject = {
...
+  name: string;
+  shader: string;
+  shaderData: {
+    [key: string]: {
+      type: "Vector2" | "Vector3" | "Vector4" | "Color" | "Float" | "Texture";
+      value: any;
```
Comment (GuoLei1990) [inline packages/loader/src/resource-deserialize/resources/prefab/PrefabDesign.ts:96]: `any` ->`Array<number> | number | IAssetRef;`

File: packages/loader/src/MaterialLoader.ts [hunk 7]
```diff
@@ -1,52 +1,39 @@
...
   ResourceManager,
   Shader,
   Texture2D,
-  UnlitMaterial
+  resourceLoader
 } from "@galacean/engine-core";
 import { Color, Vector2, Vector3, Vector4 } from "@galacean/engine-math";
+import { IPrefabMaterial } from "./resource-deserialize/resources/prefab/PrefabDesign";
+
+function setProperty(object: Object, key: string, value: any) {
```
Comment (GuoLei1990) [inline packages/loader/src/MaterialLoader.ts:15]: `setProperty`->`parseProperty`

Comment (zhuxudong) [reply packages/loader/src/MaterialLoader.ts:15]: +1

File: packages/loader/src/resource-deserialize/resources/prefab/PrefabDesign.ts (hunk 5 above)
Comment (zhuxudong) [reply packages/loader/src/resource-deserialize/resources/prefab/PrefabDesign.ts:90]: +1

File: packages/loader/src/resource-deserialize/resources/prefab/PrefabDesign.ts (hunk 6 above)
Comment (zhuxudong) [reply packages/loader/src/resource-deserialize/resources/prefab/PrefabDesign.ts:96]: +1

File: packages/loader/src/MaterialLoader.ts (hunk 4 above)
Comment (zhuxudong) [reply packages/loader/src/MaterialLoader.ts:39]: +1

File: packages/loader/src/resource-deserialize/resources/prefab/PrefabDesign.ts (hunk 2 above)
Comment (zhuxudong) [reply packages/loader/src/resource-deserialize/resources/prefab/PrefabDesign.ts:100]: +1

**Review by GuoLei1990 — APPROVED** [review]

**Pull request merged**